### PR TITLE
adds delete challenge query

### PR DESCRIPTION
**Pull request by sophielevens**
adds delete challenge query 
I've left comments where changes to the code have _actually_ been made - the rest is just linting issues

### CODE REVIEW

File: lib/queries/deleteChallenge.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+module.exports = function (id) {
```
Comment (sophielevens) [inline lib/queries/deleteChallenge.js]: delete challenge query

File: lib/queries/index.js [hunk 2]
```diff
@@ -26,6 +26,7 @@ var addTags = require('./addTags.js');
 var getOwnerChallenge = require('./getOwnerChallenge.js');
 var getUsersChallenges = require('./getUsersChallenges.js');
 var challengeEditable = require('./challengeEditable.js');
+var deleteChallenge = require('./deleteChallenge.js');
```
Comment (sophielevens) [inline lib/queries/index.js]: requiring delete challenge query

File: lib/queries/index.js [hunk 3]
```diff
@@ -53,5 +54,6 @@ module.exports = {
   addLocations: addLocations,
   getOwnerChallenge: getOwnerChallenge,
   getUsersChallenges: getUsersChallenges,
-  challengeEditable
+  challengeEditable,
```
Comment (sophielevens) [inline lib/queries/index.js]: exporting delete challenge query

File: lib/fixtures/create-tables.sql [hunk 4]
```diff
@@ -12,7 +12,7 @@ CREATE TABLE IF NOT EXISTS challenges (
 -- Create table
 CREATE TABLE IF NOT EXISTS tags_challenges (
   tags_id INTEGER,
-  challenges_id INTEGER REFERENCES challenges (id)
+  challenges_id INTEGER REFERENCES challenges (id) on delete cascade on update cascade
 );
```
Comment (Danwhy) [inline lib/fixtures/create-tables.sql]: allows challenges with tags to be deleted

File: lib/queries/deleteChallenge.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  return 'DELETE FROM challenges'
+    + ' WHERE challenges.id IN '
+    + ' (SELECT challenges.id FROM challenges'
+    + ' FULL JOIN comments '
+    + ' ON comments.challenge_id = challenges.id'
+    + ' WHERE challenges.id = ' + id
+    + ' GROUP BY challenges.id'
+    + ' HAVING COUNT(comments) = 0 );';
```
Comment (SimonLab) [inline lib/queries/deleteChallenge.js]: :+1:

**Pull request merged**